src/blocks/weather.rs

```diff
@@ -185,24 +187,80 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
         WeatherService::OpenWeatherMap(config) => Box::new(open_weather_map::Service::new(config)),
     };
 
-    loop {
-        let data = api
-            .recoverable(|| async {
-                let location = match config.autolocate {
-                    true => find_ip_location().await?,
-                    false => None,
-                };
-                provider.get_weather(location).await
-            })
-            .await?;
+    if config.autolocate {
+        // Default behavior is to mirror `interval`
+        if config.autolocate_interval.is_none() {
+            config.autolocate_interval = Some(config.interval);
+        }
+
+        if config.autolocate_interval == Some(config.interval) {
+            // In the case where `autolocate_interval` matches `interval` merge both actions.
+            loop {
+                let location: Option<Coordinates> = api.recoverable(find_ip_location).await?;
+                let data = api
+                    .recoverable(|| async { provider.get_weather(location).await })
+                    .await?;
+                widget.set_icon(data.icon.to_icon_str())?;
+                widget.set_values(data.into_values());
+                api.set_widget(&widget).await?;
+
+                select! {
+                    _ = sleep(config.interval.0) => (),
+                    _ = api.wait_for_update_request() => (),
+                }
+            }
+        } else {
+            // Two timers, one to rerender the block and the other to update the location.
+            let mut interval = tokio::time::interval(config.interval.0);
+            let mut autolocate_interval =
+                tokio::time::interval(config.autolocate_interval.unwrap().0);
+            let mut location: Option<Coordinates> = api.recoverable(find_ip_location).await?;
 
-        widget.set_icon(data.icon.to_icon_str())?;
-        widget.set_values(data.into_values());
-        api.set_widget(&widget).await?;
+            loop {
+                select! {
+                    biased; // both timers instantly `tick()` autolocate should always run first
```

Comment:
Currently, with `autolocate_interval = "once"` autolocate function is actually called twice. This is because, as you mentioned, both timers instantly tick. This can be solved by delaying the initial tick by one interval.

Also default ["missed tick behavior"](https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html) is not the best one.

So I would recommend just pre-computing both the autolocate and weather before the loop, and then create intervals using `config.interval.timer()` (it will delay the first tick and set missed tick behavior).